### PR TITLE
Allow manifest namespace to be configurable

### PR DESCRIPTION
**Pull request by rthallisey**
The manifests hardcode the kube-system namespace.  Make it configurable.

One thing I'm worried about in ```pkg/virt-controller/leaderelectionconfig/config.go``` is the ```DefaultNamespace``` var is set to kube-system.  Can folks advise on that?

```go
const (
        DefaultNamespace     = "kube-system"
)
```

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by rmohr** [comment]
> One thing I'm worried about in pkg/virt-controller/leaderelectionconfig/config.go is the DefaultNamespace var is set to kube-system. Can folks advise on that?

I think we can look for `/var/run/secrets/kubernetes.io/serviceaccount/namespace` to get the namespace it is running in. If it is not specified, we can fall-back to "kube-system". Adding a  flag would also make sense.

So the order would be as following:

 1. If flag specified, take it from the flag (useful for debugging when you run the controller outside of the cluster)
 2. If  `/var/run/secrets/kubernetes.io/serviceaccount/namespace` exists, take the namespace from there.
 3. Fall back to kube-system.

Implementing 2 and 3 would be good enough for me right now.

File: pkg/virt-controller/watch/application.go [hunk 1]
```diff
@@ -147,8 +148,16 @@ func (vca *VirtControllerApp) Run() {
 		golog.Fatalf("unable to get hostname: %v", err)
 	}
 
+	// TODO: Replace leaderelectionconfig.DefaultNamespace with a flag
+	namespace := leaderelectionconfig.DefaultNamespace
+	if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
```
Comment (rmohr) [inline pkg/virt-controller/watch/application.go]: We should check if the error is `os.IsNotExist(err)`. If it is any other error we need to exit.

**Comment by rthallisey** [comment]
@rmohr any more comments?

**Comment by davidvossel** [comment]
ok to test

**Pull request merged**